contracts/RelayedDocumentStore.sol

```diff
@@ -0,0 +1,112 @@
+pragma solidity 0.5.12;
+
+import "./DocumentStore.sol";
+import "@openzeppelin/contracts-ethereum-package/contracts/cryptography/ECDSA.sol";
+
+/**
+ * @dev This contract enables meta-transactions to be sent and delegated to a third party.
```

Comment:
i'm actually a bit confused by the purpose of this MR -

the relayer is also the owner of the document store which means he can actually call issue() without the permission of the trustedSigner - thus defeating the purpose of involving the trustedSigner

i suspect, for this to work in the way you intend - you need to create a relay contract that deploys a document store with the trustedSigner as the owner, and accepts calls from the relayer to issue & revoke on behalf of the trustedSigner

Reply:
Ah yes agreed, excellent point. Let me refactor to this pattern. Instead of inheriting the DocumentStore the Relayer will create a DocumentStore of which the contract not the relayer EOA will be the owner.

Reply:
this architecture while workable, has the issue where the trustedSigner is locked in a relationship with the relayer - if the relayer chooses to deny service he is SOL

it's a workable model but not how I would do it...
In any case if you wish to stick with this model that's fine too, but since it doesn't require modification of our DocumentStore and merely imports it, it can be done in a standalone repo that you own